ols/app/models/config.py

```diff
@@ -361,8 +391,23 @@ def set_provider_specific_configuration(self, data: dict) -> None:
                     azure_config = data.get("azure_openai_config")
                     self.check_provider_config(azure_config)
                     if azure_config is not None:
-                        azure_config["client_secret"] = _get_attribute_from_file(
-                            azure_config, "client_secret_path"
+                        azure_config["tenant_id"] = _read_secret(
+                            azure_config,
+                            constants.CREDENTIALS_PATH_SELECTOR,
+                            constants.AZURE_TENANT_ID_FILENAME,
+                            raise_on_error=False,
+                        )
+                        azure_config["client_id"] = _read_secret(
+                            azure_config,
+                            constants.CREDENTIALS_PATH_SELECTOR,
+                            constants.AZURE_CLIENT_ID_FILENAME,
+                            raise_on_error=False,
+                        )
+                        azure_config["client_secret"] = _read_secret(
+                            azure_config,
+                            constants.CREDENTIALS_PATH_SELECTOR,
+                            constants.AZURE_CLIENT_SECRET_FILENAME,
+                            raise_on_error=False,
```

Comment:
i think there is a flaw here.....if someone sets "credentials_path" to "/foo/bar/somefilename", then "tenant_id", "client_id", and "client_secret" are all going to get populated w/ the value of whatever is in "somefilename", no?

and then "api_key" will also get set to the filename value.  Which i guess means we'll prefer to use the "api_key" value as the credentials, so maybe it doesn't matter that much?  But it still seems bad that we'd populate the other dictionary keys from that file.

Reply:
indeed, you are right. Jmm ok, it would be possible to have keyword flag "must_be_directory" perhaps (just for Azure OpenAI-related secret).

Reply:
`azure_openai_config` section is new and OLS can expect  `credentials_path` to be a directory 
(1) Fail the validation if it IS a file 
else (2) check the directory for the presence of `apitoken` file , if `apitoken`  is present use that as credential 
else (3) If `apitoken` check for `tenant_id` `client_id` and `client_secret` - use them if all 3 are present.
else (4) Fail the credential validation. 
From the operator stand point , we are only setting a directory - and the previous versions of operator we did not have this `azure_openai_config` section.

Reply:
@bparees the flaw should be fixed now.
@xrajesh I guess it's there now (I guess that by [3] you mean if `not apitoken`